inline
unordered_[set]
exception("") -- wrong